### DISCUSSION TITLE
Splash screen error "Failed to load tcl/tk libraries" on test system, but not on dev system

### DISCUSSION THREAD

**Discussion by mhucka** [pyinstaller/pyinstaller#6542] [Help] [answered]
I've used PyInstaller in one-file mode for many open-source applications, but only recently tried to use a splash screen for a Windows app. The app with a splash screen works as expected on my Windows 10 build system, but when I copy the app to another Windows 10 system (i.e., a completely separate VM), it fails early because it is **unable to find the tcl/tk libraries** at run time – **even though the `.dll` files appear to be there**. Here is a screen showing the output with `debug` = True and `console = True`:

![windows-splashscreen-error](https://user-images.githubusercontent.com/1450019/151280632-133d02e3-66ad-42d6-b3e6-e7a896c32c5f.png)

When I look in the temp directory for the files listed in the last 3 lines shown above, **the files exist in those locations**. Here is what that directory looks like:

![image](https://user-images.githubusercontent.com/1450019/151280817-301b7390-a81b-4a0d-b861-1fa62413cb41.png)

If I remove the Splash screen parts of the PyInstaller configuration and build the app without the splash screen, the error does not occur and the application runs normally on Windows 10 test systems. If I build in one-dir mode, the error also does not occur.

Does anyone recognize these symptoms? Can anyone suggest avenues to explore?

## More info about the app

* PyInstaller spec file: <https://github.com/caltechlibrary/foliage/blob/main/pyinstaller-win32.spec>
* Script that I use to build the application: <https://github.com/caltechlibrary/foliage/blob/main/make.bat>
* The problem occurs whether I use PyInstaller 4.8 or the latest 5.0 dev version.
* The PyInstaller application is built in one-file mode

## Answers to basic questions:

* Output of `pyinstaller --version`: 4.8
* Version of Python: 3.9.7
* Platform: Windows 10
* How you installed Python: using python.org/downloads

I have done the following:
  * [x] start with clean installation
  * [x] use the latest development version
  * [x] Run your frozen program **from a command window (shell)** — instead of double-clicking on it
  * [x] Package your program in **--onedir mode** (_works in one-dir mode_)
  * [x] Package **without UPX**, say: use the option `--noupx` or set `upx=False` in your .spec-file
  * [x] Repackage application in **verbose/debug mode**. For this, pass the option `--debug` to `pyi-makespec` or `pyinstaller` or use `EXE(..., debug=1, ...)` in your .spec file.

**Comment by rokm** [comment, marked as answer]
Duplicate of #6284. The splash screen is shown before the onefile build is fully unpacked and due to an oversight, at that point the dependencies of Tcl/Tk shared libraries (i.e., `vcruntime140.dll`) are not unpacked, leading to the error if the target system does not have its own copy of `vcruntime140.dll` installed.

**Reply by mhucka** [reply]
Thank you for recognizing this problem! I'm glad I'm not doing something dumb. I did look through past issues but evidently not carefully enough …

Is there a workaround by any chance? Issue #6284 mentions "need to add binary dependencies of Tcl/Tk shared libs to list of splash screen requirements so that they get extracted before we try to initialize the splash screen", which sounds like something I could do via the spec file, but I'm not sure how …

**Reply by bwoodsend** [reply]
No, I'm afraid that the only way around this is to fix the issue. The next closest you can get is to tell your users to install `vcruntime140.dll` manually.
